Editor/Scripts/Analytics/MeshsyncAnalytics.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+using UnityEditor;
+
+using UnityEngine;
+
+namespace Unity.MeshSync.Editor.Analytics {
+
+    internal static class MeshSyncAnalyticsFactory {
+
+        public static IMeshSyncAnalytics CreateAnalytics() {
+            return new MeshSyncAnalytics();
+        }
+    }
+
+    internal interface IMeshSyncAnalytics {
+
+        /// <summary>
+        /// Send an event that a user installed and configured a DCC tool for use with MeshSync
+        /// </summary>
+        /// <param name="dccTool">Short designation for the DCC which was installed and configured</param>
+        void UserInstalledPlugin(string dccTool);
+
+        void UserSyncedData(MeshSyncAnalyticsData data);
+    }
+
+    internal sealed class MeshSyncAnalytics : IMeshSyncAnalytics {
+        private const string VENDORKEY = "unity.meshsync";
+
+        private const string DCCCONFIGURE_INSTALLEDEVENTNAME = "dccConfigure";
+        private const int DCCCONFIGURE_VERSION = 4;
+
+        private const string MESHSYNC_SYNC_INSTALLEDEVENTNAME = "meshSync_Sync";
+        private const int MESHSYNC_SYNC_VERSION = 2;
```

Comment:
What does version mean in this context? I see its 2 for this event and 4 for DCCCONFIGURE. Where are these values derived from?

Reply:
So the versions here refer specifically to the schema version, the 4 on line 33 is used to construct the schema string that we're publishing data to: com.unity3d.data.schemas.editor.analytics.meshSync_Sync_v2. I went through a few revisions of schemas before I got the data we needed in the API hence why we have version 4 for the dccInstall event and  version 2 for meshSync_Sync event.